### PR TITLE
Added Lessons 2,3,4,5,6,8,9,10. 9 and 10 need a little modification

### PR DESCRIPTION
**Pull request by Charith-Acharya**

### CODE REVIEW

File: students/Charith_Acharya1/Circle_excercise.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        return self.radius * 2
+
+    @diameter.setter
+    def diameter(self, value):
+        self.radius = value / 2
+
+    @property
+    def circle_area(self):
+         self.area = math.pi * self.radius * self.radius
+         return self.area
```
Comment (natasha-aleksandrova) [inline students/Charith_Acharya1/Circle_excercise.py]: Per instructions: "The user should not be able to set the area:"

File: students/Charith_Acharya1/Circle_excercise.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    @property
+    def circle_area(self):
+         self.area = math.pi * self.radius * self.radius
+         return self.area 
+     
+    @classmethod
+    def circle_from_diameter(cls, diameter):
+        cls.area_dia = math.pi * diameter/2 * cls.diameter/2
+        return cls.area_dia
```
Comment (natasha-aleksandrova) [inline students/Charith_Acharya1/Circle_excercise.py]: method is supposed to be called `from_diameter` and supposed to return an instance of Circle

File: students/Charith_Acharya1/Lesson3.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+###Lesson 3        
+##Slicing Lab Excercise
+
+#First and last items exchanged
+def first_last_exchange(seq):
+    first = seq[0]
+    middle = seq[1:len(seq)-1]
+    last = seq[-1]
+    result  = (last,middle,first)
+    print(result)
```
Comment (natasha-aleksandrova) [inline students/Charith_Acharya1/Lesson3.py]: this needs to use slicing only (no indexing)

File: students/Charith_Acharya1/Lesson3.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+#Every other item removed
+def first_last_only(seq):
+    first = seq[0]
+    middle = seq[1:len(seq)-1]
+    last = seq[-1]
+    result  = (first,last)
+    print(result)
```
Comment (natasha-aleksandrova) [inline students/Charith_Acharya1/Lesson3.py]: same here, slicing only

File: students/Charith_Acharya1/Lesson3.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+#First four and last four items removed and then every other item inbetween
+def four(seq):
+    f = seq[0:3]
+    l = seq[-1:-4]
+    newseq = seq[4:len(seq)-4]
+    finalseq = ""
+    for i in range(len(newseq)):
+        if i%2==0:
+            finalseq = finalseq+ newseq[i]
+    print(finalseq)
```
Comment (natasha-aleksandrova) [inline students/Charith_Acharya1/Lesson3.py]: same here, slicing only

File: students/Charith_Acharya1/Lesson3.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+#With the elements reversed
+def reverse(seq):
+    negseq = seq[::-1]
+    print(negseq)
```
Comment (natasha-aleksandrova) [inline students/Charith_Acharya1/Lesson3.py]: missing another slicing function.

File: students/Charith_Acharya1/Lesson6.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        SwitchFuncDict.get(response)()
+
+def Thanks():
+    Full_Name = input("Please enter the full name".title())
+    New_Amount = input("Please enter new amount")
+    Amount = int(New_Amount)
+    if Full_Name in Master_List.keys():
+        Master_List[Full_Name].append(Amount)
+        print(Message(Full_Name, New_Amount))        
+    if Full_Name not in Master_List.keys():
```
Comment (natasha-aleksandrova) [inline students/Charith_Acharya1/Lesson6.py]: needs to be an `else` statement

File: students/Charith_Acharya1/Lesson6.py [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+        Len = len(Master_List[i])
+        Average = Sum/Len
+        Sum_List.append(Sum)
+        Len_List.append(Len)
+        Avg_List.append(Average)            
+    print(Sum_List)
+    print(Len_List)
+    print(Avg_List)
+    for i in range(len(list(Master_List.keys()))):
+        print(list(Master_List.keys())[i] + "-"*5 + str(Sum_List[i]) + "-"*5 + str(Len_List[i]) +"-"*5 + str(Avg_List[i]))
```
Comment (natasha-aleksandrova) [inline students/Charith_Acharya1/Lesson6.py]: missing sorting feature

File: students/Charith_Acharya1/Lesson9.py [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+        try:
+            self.donations.append(self.amount)
+        except ValueError:
+            print("Enter a donation amount")
+            
+    def total_donations(self):
+        try:
+            return sum(self.donations)
+        except TypeError:
+            return self.donations
```
Comment (natasha-aleksandrova) [inline students/Charith_Acharya1/Lesson9.py]: why return donations in a case of error?

**Comment by Charith-Acharya** [comment]
Hello,
Thank you for your comments on my assignments. I am sorry for the
inconvenience posed by submitting all of them at one time. I  was using my
official laptop for my assignments and it was blocking my access to GitHub.
I do not have a personal laptop and had to use someone else’s laptop  to
submit my assignments.

I see that my course ends on Apr 17. May I get some additional time to make
the corrections? This is my first attempt at programming and it took me
considerable effort and extra time to complete the assignments.

Would appreciate it if I get some additional time to make the corrections
and submit, please let me know!

Regards,
Charith Acharya

**Comment by natasha-aleksandrova** [comment]
Hi Charith - thanks for reaching out, please email us with a request for extension before your deadline

**Pull request merged**